flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimV1.java

```diff
@@ -84,6 +84,7 @@ public Function getFunction(IMetaStoreClient client, String dbName, String funct
 			// hive-1.x doesn't throw NoSuchObjectException if function doesn't exist, instead it throws a MetaException
 			return client.getFunction(dbName, functionName);
 		} catch (MetaException e) {
+			// need to check the cause and message of this MetaException to decide whether it should actually be a NoSuchObjectException
```

Comment:
Is this specific to HIve 1.2.1? If so, let's be clear, since we didn't have this check previously.

Reply:
Yeah it's specific to 1.2.1. Actually there's a comment about that a couple of lines above:
```
// hive-1.x doesn't throw NoSuchObjectException if function doesn't exist, instead it throws a MetaException
```